Sources/Examples/ExploringPerformanceExample.swift

```diff
@@ -0,0 +1,201 @@
+import Foundation
+import Logging
+
+// MARK: Example library code
+
+struct DataContainer {
+    let data: String // this would be some user `T` type
+
+    // would be W3C's `traceparent` for example
+    let traceContext: TraceContext?
+}
```

Comment:
This is just an example, though showcases a series of cases where "we want to carry around a nice binary representation and only render if when we really need to", in this case it could be a trace context from which we'd for example only want to log the `traceId` for correlation purposes of spans with logs.

(yes, tracing is a separate domain entirely, however correlating with logging is very useful, and since "tons of things" would have this, it would be good to not pay pentalties for it when we don't actually log with them).